src/System.Private.CoreLib/src/System/Collections/Generic/ComparerHelpers.cs

```diff
@@ -50,7 +50,7 @@ internal static object CreateDefaultComparer(Type type)
             // The comparer for enums is specialized to avoid boxing.
             else if (type.IsEnum)
             {
-                result = TryCreateEnumComparer(runtimeType);
+                result = CreateInstanceForAnotherGenericParameter((RuntimeType)typeof(EnumComparer<DayOfWeek>), runtimeType);
```

Comment:
Random question -- why is 'DayOfWeek' passed in as the type arg here?

Reply:
I just picked an enum since the other calls to `CreateInstanceForAnotherGenericParameter` specify a sample type argument.

Reply:
The sample type argument is not required. The other calls to `CreateInstanceForAnotherGenericParameter` specify a sample type argument if that instantiation is very likely used by a typical program. I do not think it is the case here. `EnumComparer<DayOfWeek>` is unlikely to be used by a typical program.